clar2wasm/src/wasm_generator.rs

```diff
@@ -1791,9 +1270,6 @@ impl WasmGenerator {
             TypeSignature::IntType => "int",
             TypeSignature::UIntType => "uint",
             _ => {
-                self.error = Some(GeneratorError::InternalError(
-                    "invalid type for shift".to_string(),
-                ));
                 return Err(GeneratorError::NotImplemented);
```

Comment:
Can this return the `InternalError` instead of the `NotImplemented` error?

There are several other places where you've dropped the additional info in the internal error and just left the "not implemented" error.

Reply:
addressed, along with all similar cases